### ISSUE TITLE
Compilation warnings with gcc for vsearch 2.10

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/vsearch#350]
(tested versions: `gcc` 4.9 5.1 5.2 5.3 5.4 5.5 6.1 6.2 6.3 6.4 6.5 7.1 7.2 7.3 7.4 8.1 8.2)

Surprisingly, there is a warning with `gcc` 5.1, but not with other `gcc` 5 versions:
```
gcc (GCC) 5.1.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

chimera.cc: In function 'uint64_t chimera_thread_core(chimera_info_s*)':
chimera.cc:1239:32: warning: array subscript is above array bounds [-Warray-bounds]
             if (ci->cand_list[k] == target)
```

Compilation warnings with `gcc` 8.1 and 8.2:
```
gcc (GCC) 8.2.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

In function 'void pushop(s16info_s*, char)',
    inlined from 'void backtrack16(s16info_s*, char*, uint64_t, uint64_t, uint64_t, short unsigned int*, short unsigned int*, short unsigned int*, short unsigned int*)' at align_simd.cc:994:13:
align_simd.cc:855:18: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 10 [-Wstringop-truncation]
           strncpy(s->cigarend, buf, len);
           ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In function 'void pushop(s16info_s*, char)',
    inlined from 'void backtrack16(s16info_s*, char*, uint64_t, uint64_t, uint64_t, short unsigned int*, short unsigned int*, short unsigned int*, short unsigned int*)' at align_simd.cc:1011:13:
align_simd.cc:855:18: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 10 [-Wstringop-truncation]
           strncpy(s->cigarend, buf, len);
           ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In function 'void pushop(s16info_s*, char)',
    inlined from 'void backtrack16(s16info_s*, char*, uint64_t, uint64_t, uint64_t, short unsigned int*, short unsigned int*, short unsigned int*, short unsigned int*)' at align_simd.cc:1001:13:
align_simd.cc:855:18: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 10 [-Wstringop-truncation]
           strncpy(s->cigarend, buf, len);
           ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In function 'void pushop(s16info_s*, char)',
    inlined from 'void backtrack16(s16info_s*, char*, uint64_t, uint64_t, uint64_t, short unsigned int*, short unsigned int*, short unsigned int*, short unsigned int*)' at align_simd.cc:1021:13:
align_simd.cc:855:18: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 10 [-Wstringop-truncation]
           strncpy(s->cigarend, buf, len);
           ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In function 'void pushop(s16info_s*, char)',
    inlined from 'void backtrack16(s16info_s*, char*, uint64_t, uint64_t, uint64_t, short unsigned int*, short unsigned int*, short unsigned int*, short unsigned int*)' at align_simd.cc:1030:13:
align_simd.cc:855:18: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 10 [-Wstringop-truncation]
           strncpy(s->cigarend, buf, len);
           ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In function 'void finishop(s16info_s*)',
    inlined from 'void backtrack16(s16info_s*, char*, uint64_t, uint64_t, uint64_t, short unsigned int*, short unsigned int*, short unsigned int*, short unsigned int*)' at align_simd.cc:1033:11:
align_simd.cc:872:18: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 10 [-Wstringop-truncation]
           strncpy(s->cigarend, buf, len);
           ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by torognes** [comment]
The warnings in gcc version 8 are fixed in vsearch 2.10.3. I think the warning with gcc 5.1.0 must be a compiler bug.

**Comment by frederic-mahe** [comment]
No more warnings with gcc 8! I agree that the gcc 5.1 warning is a bug.
